src/proto/project.proto

```diff
@@ -0,0 +1,85 @@
+  PROJECT_STATUS_ACTIVE = 1;   // Active project
+}
+
+enum ProjectAssetUse {
+  PROJECT_ASSET_USE_INPUT = 0;        // Asset is used as an input for the project
+  PROJECT_ASSET_USE_OUTPUT = 1;       // Asset is an output from the project
+  PROJECT_ASSET_USE_INTERMEDIATE = 2; // Asset is an output from an intermediate step of the project
+  PROJECT_ASSET_USE_PROJECT = 3;      // Asset is used as the project file for the project
+}
+
+message ProjectAsset {
+  uint64 id = 1;           // ID of the asset
+  string URI = 2;          // The URI to the asset
+  ProjectAssetUse use = 3; // The use of the asset with respect to the project
+}
+
+message Project {
+  uint64 id = 1;                     // ID of the project
+  string title = 2;                  // Title of the project
+  string owner = 3;                  // The ID of the organization owning the project
+  ProjectAccess access = 4;          // The access of the project (such as public vs private)
+  ProjectStatus status = 5;          // The status of the project (such as active vs inactive)
+  uint64 size = 6;                   // The size of the project in bytes
+  ProjectAsset file = 7;             // Project file for the project
+  uint64 created_at = 8;             // Time of project creation (UTC seconds epoch)
+  uint64 updated_at = 9;             // Time of last project update (UTC seconds epoch)
+  repeated ProjectAsset assets = 10; // The listing of all assets associated with the project
+}
+
+message CreateProjectRequest {
+  string title = 1;               // title of the project
+  optional string owner = 2;      // The ID of the organization owning the project, empty for default org for user
+  ProjectAccess access = 3;       // The access of the project (such as public vs private)
+  ProjectStatus status = 4;       // The status of the project (such as active vs inactive)
+  optional ProjectAsset file = 5; // Project file for the project (if existing, else will create)
+}
+
+message UpdateProjectRequest {
+  uint64 id = 1;                     // ID of the project
+  optional string owner = 2;         // The ID of the organization owning the project, empty for default org for user
+  optional string title = 3;         // Title of the project
+  optional ProjectAccess access = 4; // The access of the project (such as public vs private)
+  optional ProjectStatus status = 5; // The status of the project (such as active vs inactive)
+  optional ProjectAsset file = 6;    // Project file for the project
+}
+
+message ListProjectRequest {
+  optional string owner = 1; // The ID of the organization owning the project, empty for default org for user
+}
+
+message GetDeleteProjectRequest {
+  uint64 id = 1;             // ID of the project to request
+  optional string owner = 2; // The ID of the organization owning the project, empty for default org for user
+}
+
+// Message for empty requests / response where no additional data is needed
+message ProjectEmpty {
```

Comment:
What I've found useful is to return the data structure being deleted, but with a `deleted` status to communicate that X was deleted.

Reply:
I don't have any issue with returning it.